src/language/visitor.ts

```diff
@@ -371,113 +294,99 @@ export class Visitor {
+          this._checkIsArrayWithinSelection(item);
+        }
+      });
+    }
   }
 
-  _checkIsSelectionInVariable(node: babel.types.VariableDeclarator) {
+  _checkIsBSONSelectionInVariable(node: babel.types.Node) {
     if (
+      node.type === 'VariableDeclarator' &&
       node.init &&
-      (
-        node.init.type === 'ObjectExpression' ||
-        node.init.type === 'ArrayExpression'
-      ) &&
-      this._checkIsVariableIdentifierBeforeSelection(node) &&
-      this._checkIsWithinSelection(node.init)
+      this._isVariableIdentifierBeforeSelection(node)
     ) {
-      return true;
+      this._checkIsObjectWithinSelection(node.init);
+      this._checkIsArrayWithinSelection(node.init);
     }
-
-    return false;
   }
 
-  _checkIsSelectionInObject(node: babel.types.ObjectProperty) {
+  _checkIsBSONSelectionInObject(node: babel.types.Node) {
     if (
+      node.type === 'ObjectProperty' &&
       node.value &&
-      (
-        node.value.type === 'ObjectExpression' ||
-        node.value.type === 'ArrayExpression'
-      ) &&
-      this._checkIsObjectPropBeforeSelection(node) &&
-      this._checkIsWithinSelection(node.value)
+      this._isObjectPropBeforeSelection(node)
     ) {
-      return true;
+      this._checkIsObjectWithinSelection(node.value);
+      this._checkIsArrayWithinSelection(node.value);
     }
-
-    return false;
   }
 
-  _checkIsBSONSelection(node: babel.types.Node): boolean {
-    if (node.type === 'CallExpression') {
-      return this._checkIsSelectionInFunction(node);
-    }
-
-    if (node.type === 'ArrayExpression') {
-      return this._checkIsSelectionInArray(node);
-    }
-
-    if (node.type === 'VariableDeclarator') {
-      return this._checkIsSelectionInVariable(node);
-    }
-
-    if (node.type === 'ObjectProperty') {
-      return this._checkIsSelectionInObject(node);
-    }
-
-    return false;
+  _checkIsBSONSelection(node: babel.types.Node): void {
+    this._checkIsBSONSelectionInFunction(node);
+    this._checkIsBSONSelectionInArray(node);
+    this._checkIsBSONSelectionInVariable(node);
+    this._checkIsBSONSelectionInObject(node);
   }
 
-  _checkIsCollectionName(node: babel.types.CallExpression | babel.types.MemberExpression): boolean {
+  _checkIsCollectionNameAsMemberExpression(node: babel.types.Node): void {
     if (
-      (node.type === 'MemberExpression' &&
-        node.object.type === 'Identifier' &&
-        node.object.name === 'db' &&
-        node.property.type === 'Identifier' &&
-        node.property.name.includes(PLACEHOLDER)) ||
-      (node.type === 'CallExpression' &&
-        node.callee.type === 'MemberExpression' &&
-        this._checkIsCollectionName(node.callee))
+      node.type === 'MemberExpression' &&
+      node.object.type === 'Identifier' &&
+      node.object.name === 'db' &&
+      node.property.type === 'Identifier' &&
+      node.property.name.includes(PLACEHOLDER)
     ) {
-      return true;
+      this._state.isCollectionName = true;
```

Comment:
Are these variables ever unset or set to `false`?

Reply:
Yeah, they all are being reset with each parse of the file:
https://github.com/mongodb-js/vscode/blob/apply-tell-do-not-ask-principle/src/language/visitor.ts#L141